commit 28fac3973b3c693c453594b007c08c1e1e5c4af3
Author: Ilya Kuzmich <ilya.kuzmich@gmail.com>
Date: Tue, 9 Jan 2018 14:37:43 +0300

    merge_config.sh: add br2-external support

    Pass BR2_EXTERNAL value via -e option.
    This will prevent merge_config.sh from silently eating any symbols defined in
    external trees on a clean buildroot tree invocation.

    Signed-off-by: Ilya Kuzmich <ilya.kuzmich@gmail.com>
    Acked-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
    Signed-off-by: Peter Korsgaard <peter@korsgaard.com>

Index: kconfig/merge_config.sh
===================================================================
--- kconfig.orig/merge_config.sh
+++ kconfig/merge_config.sh
@@ -33,6 +33,7 @@ usage() {
 	echo "  -n    use allnoconfig instead of alldefconfig"
 	echo "  -r    list redundant entries when merging fragments"
 	echo "  -O    dir to put generated output files.  Consider setting \$KCONFIG_CONFIG instead."
+	echo "  -e    colon-separated list of br2-external trees to use (optional)"
 }
 
 RUNMAKE=true
@@ -71,6 +72,11 @@ while true; do
 		shift 2
 		continue
 		;;
+	"-e")
+		EXTERNAL_ARG="BR2_EXTERNAL=$2"
+		shift 2
+		continue
+		;;
 	*)
 		break
 		;;
@@ -151,7 +157,7 @@ fi
 # Use the merged file as the starting point for:
 # alldefconfig: Fills in any missing symbols with Kconfig default
 # allnoconfig: Fills in any missing symbols with # CONFIG_* is not set
-make KCONFIG_ALLCONFIG=$TMP_FILE $OUTPUT_ARG $ALLTARGET
+make KCONFIG_ALLCONFIG=$TMP_FILE $EXTERNAL_ARG $OUTPUT_ARG $ALLTARGET
 
 
 # Check all specified config values took (might have missed-dependency issues)
